### ISSUE TITLE
Support progressive forward seeking (without flushing the decoder)

### ISSUE THREAD

**Issue by ZeroOneZeroR** [google/ExoPlayer#9449]
Suppose my current player position is 100ms. Now I want to seek to 150ms. As 150ms is slightly greater than 100ms, we don't need to seek to previous sync frame and come to current seek position. We can just step forward the decoder. I think this will be useful in smooth and fast forward seeking. Is this possible in this player? If it is not possible directly, how can I achieve this? Any thoughts would be appreciated.

**Comment by krocard** [comment]
I don't believe this is currently supported by ExoPlayer. It would need to special case a transition that doesn't go over a sync frame.
@andrewlewis do you think such optimisation would be worth tracking?

**Comment by andrewlewis** [comment]
Yes, I think it's fine to track though I'm not sure about prioritization (so I will mark it as low priority for the time being). I didn't spot an existing enhancement request for this.
